editor/plugins/tiles/atlas_merging_dialog.cpp

```diff
@@ -104,10 +106,12 @@ void AtlasMergingDialog::_generate_merged(Vector<Ref<TileSetAtlasSource>> p_atla
 				// Create tiles and alternatives, then copy their properties.
 				for (int alternative_index = 0; alternative_index < atlas_source->get_alternative_tiles_count(tile_mapping.key); alternative_index++) {
 					int alternative_id = atlas_source->get_alternative_tile_id(tile_mapping.key, alternative_index);
-					if (alternative_id == 0) {
+					if (alternative_index == 0) {
 						merged->create_tile(tile_mapping.value, atlas_source->get_tile_size_in_atlas(tile_mapping.key));
+						ERR_FAIL_COND(!merged->has_tile(tile_mapping.value));
```

Comment:
I checked if this condition helps with anything and I still get a crash without the `set_texture_region_size()` fix.